pkg/finder/finder.go

```diff
@@ -464,3 +472,23 @@ func getContainerEnvironmentVariables(ci ContainerInfo) map[string]string {
 	// nothing else to try, give up and fail!
 	return nil
 }
+
+func getAllocatedCPUs(cgroupsPath string, CRIEndpointPath string) (string, error) {
+	var cgroupAbsolutePath string
+	switch CRIEndpointPath {
+	case "/host-run/crio/crio.sock":
+		cgroupPathFromCRI := strings.Replace(cgroupsPath, "slice:crio:", "slice/crio-", 1)
```

Comment:
once we have a flag we can extract the path extraction from this function and set it explicitely in main, so the runtime-specific step is clearly separated

Reply:
This turned out very poorly written, it's probably simpler to demonstrate the concept with this (completely untested): https://gist.github.com/fromanirh/5de2ef51be8d6f3a978b8d5bf3418b5d

Reply:
So elegant! Will update the code, Thanks for the input!